tools/docker/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM alpine:latest
+RUN apk update
+RUN apk add cargo
+RUN wget https://github.com/mbrubeck/agate/archive/refs/tags/v3.0.2.tar.gz
+RUN tar xzf v3.0.2.tar.gz
+RUN ls /
+RUN mv /agate-3.0.2 /agate
+WORKDIR "/agate"
+RUN ls
+RUN cargo build --release
```

Comment:
You could use the precompiled binaries instead which should remove the compile time overhead.

Reply:
Do you think I should make an extra Dockerfile for people that have an architecture that has no precompiled binaries?

Reply:
Now I changed the Dockerfile to use the precompiled binary but I always get an '/bin/sh: /agate: not found'. Do you know what could be the problem? The file exists, it has content and is executable.

Reply:
I have no idea as I don't know much about docker. Maybe try going back to compiling from source? Since you pointed out it creates some issues. What do you think?

Reply:
Okay, I duckduckgoed a bit and it seems like it's because the precompiled binaries require glibc but alpine (what I choosed as base for the image) uses musl libc.
There are multiple options to fix this problem:
* go back to compiling from source
* use something like debian, which supports glibc, as a base
* add musl libc binaries to the release section

I would stay with compiling from source for now, because it only takes a few minutes and the alpine base is way more efficient and secure than debian.

Reply:
sounds good to me!

Reply:
I just noticed that I could make the image even more efficient by removing the source directory and cargo after compilation.

Reply:
Do I have to make one big diff, that bundles all my changes, afterwards?